### PR TITLE
feat(filter): membership operator

### PR DESCRIPTION
**Pull request by numb3r3**
add membership operator `$in` and `$nin`, e.g., 

`brand` equals Nike or Gucci using `$in` operator
```
{'brand': {'$in': ['Nike', 'Gucci']}}
```

`color` not equal white nor blue using `$nin` operator:

```
{'color': {'$nin': ['white', 'blue']}}
```

### CODE REVIEW

**Review by davidbp — APPROVED** [review]
LGTM!

File: pqlite/filter.py [hunk 1]
```diff
@@ -33,15 +35,23 @@ def _sql_parsing(data, default_logic: str = 'AND'):
...
-                parameters.append(val)
-                if i == 0:
+                if i > 0:
+                    where_clause += f' {default_logic} '
+
+                if op in COMPARISON_OPERATORS:
+                    parameters.append(val)
                     where_clause += f'({key} {COMPARISON_OPERATORS[op]} ?)'
+                elif op in MEMBERSHIP_OPERATORS:
+                    parameters.extend(val)
```
Comment (JoanFM) [inline pqlite/filter.py:50]: why extend?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/test_index.py [hunk 2]
```diff
@@ -222,3 +222,38 @@ def test_search_numpy_filter_str(
...
+    X = np.random.random((Nq, D)).astype(np.float32)
+    query_np = np.array([X[i] for i in range(Nq)])
+    da = heterogenenous_da
+
+    dists, doc_ids = pqlite_with_heterogeneous_tags.search_numpy(
+        query_np,
+        filter={'category': {'$in': ['comics', 'audiobook']}},
+        include_metadata=True,
+    )
+    for doc_ids_query_k in doc_ids:
```
Comment (JoanFM) [inline tests/test_index.py:240]: assert len of doc_ids otherwise if len doc_ids is 0 the test will give a false positive

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/test_index.py (hunk 2 above)
Comment (davidbp) [reply tests/test_index.py:244]: true, fixed

File: pqlite/filter.py (hunk 1 above)
Comment (numb3r3) [reply pqlite/filter.py:50]: the conditional value of `$in` or `$nin` is an array

**Pull request merged**